src/ui/public/courier/data_source/build_query/from_kuery.js

```diff
@@ -0,0 +1,52 @@
+import _ from 'lodash';
+import { fromKueryExpression } from '../../../kuery';
+
+function astToQuery(ast) {
+  if (Array.isArray(ast)) {
```

Comment:
I took a swing at this: https://github.com/elastic/kibana/pull/11915/commits/d1b37e5a68ef689badde93f7a27efd2ed44d7da5. For now I'm calling it a `compound` query. Let me know what you think. Keep in mind we can change this pretty easily, the only thing that gets stored in ES is the query string itself.